chain/mempool.go

```diff
@@ -10,12 +10,21 @@ import (
 	"github.com/btcsuite/btcd/btcutil"
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/btcsuite/btcd/wire"
-	"golang.org/x/sync/errgroup"
 )
 
-// txNotFoundErr is an error returned from bitcoind's `getrawtransaction` RPC
-// when the requested txid cannot be found.
-const txNotFoundErr = "-5: No such mempool or blockchain transaction"
+const (
+	// txNotFoundErr is an error returned from bitcoind's
+	// `getrawtransaction` RPC when the requested txid cannot be found.
+	txNotFoundErr = "-5: No such mempool or blockchain transaction"
+
+	// getRawTxBatchSize specifies the number of requests to be batched
+	// before sending them to the bitcoind client.
+	getRawTxBatchSize = 10_000
```

Comment:
Decreased by 10x. Not sure if the 32MB body size limit also applies for batch rpc calls? 

> Additionally, I think this might cause other goroutines calling bitcoind to have to wait a long time for the batch to finish

Probably. I assume the lookup will be fast in `bitcoind`, sth like `log(N)` if indexed. And because we are sending 10000x fewer requests here, it will actually make `bitcoind` faster to respond because of the overhead saved.

Reply:
I think any gain would come from less HTTP headers. `bitcoind` will still call the [same function N times](https://github.com/bitcoin/bitcoin/blob/744157ef1a0b61ceb714cc27c9ae158907aecdc9/src/rpc/rawtransaction.cpp#L261) and still acquire the main `cs_main` lock N times. I'd be curious to benchmark this in a reproducible way and see where any gain comes from, but not sure how feasible that is.

The 32MB body size applies for all incoming HTTP requests to bitcoind, it's set here: https://github.com/bitcoin/bitcoin/blob/744157ef1a0b61ceb714cc27c9ae158907aecdc9/src/httpserver.cpp#L457